src/transformers/pipelines/depth_estimation.py

```diff
@@ -96,18 +98,24 @@ def preprocess(self, image, timeout=None):
         return model_inputs
 
     def _forward(self, model_inputs):
-        model_outputs = self.model(**model_inputs)
-        return model_outputs
+        return self.model(**model_inputs)
 
     def postprocess(self, model_outputs):
-        predicted_depth = model_outputs.predicted_depth
-        prediction = torch.nn.functional.interpolate(
-            predicted_depth.unsqueeze(1), size=self.image_size[::-1], mode="bicubic", align_corners=False
+        outputs = self.image_processor.post_process_depth_estimation(
+            model_outputs,
+            [self.image_size[::-1]],
         )
-        output = prediction.squeeze().cpu().numpy()
-        formatted = (output * 255 / np.max(output)).astype("uint8")
-        depth = Image.fromarray(formatted)
-        output_dict = {}
-        output_dict["predicted_depth"] = predicted_depth
-        output_dict["depth"] = depth
-        return output_dict
+
+        outputs = [
+            {
+                "predicted_depth": output,
+                "depth": Image.fromarray(
+                    (((output - output.min()) / (output.max() - output.min())).detach().cpu().numpy() * 255).astype(
+                        "uint8"
+                    )
+                ),
+            }
+            for output in outputs
+        ]
+
+        return outputs[0] if len(outputs) == 1 else outputs
```

Comment:
* otherwise the returned `predicted_depth` will still be a torch tensor with grads
* more clear than trying to put everything on on-line 

```suggestion
        formatted_outputs = []
        for output in outputs:
            output = output.detach().cpu().numpy()
            
            predicted_depth = output
            depth = (output - output.min()) / (output.max() - output.min())
            depth = (depth * 255).astype("unit8")

            formatted_outputs.append({"predicted_depth": output, "depth": depth})

        return formatted_outputs[0] if len(outputs) == 1 else formatted_outputs
```

Reply:
I did not convert the `predicted_depth` to a numpy array in order to match the input type as much as possible (device, gradients, etc.). If we do not want this, then I agree, otherwise, I believe we should not run "mess" with the type of the data going to `predicted_depth`

Reply:
Agreed - let's avoid the numpy conversion